src/axolotl/utils/models.py

```diff
@@ -83,37 +85,47 @@ def load_model(
     adapter="lora",
     inference=False,
 ):
-    # type: (str, str, str, str, DictDefault, Optional[str], bool) -> Tuple[PreTrainedModel, Optional[PeftConfig]]
+    # type: (str, str, str, AutoTokenizer, DictDefault, Optional[str], bool) -> Tuple[PreTrainedModel, Optional[PeftConfig]]
     """
     Load a model from a base model and a model type.
     """
 
     # TODO refactor as a kwarg
     load_in_8bit = cfg.load_in_8bit
-    is_llama_derived_model = "llama" in base_model or (
+    cfg.is_llama_derived_model = "llama" in base_model or (
         cfg.model_type and "llama" in cfg.model_type.lower()
     )
 
-    if is_llama_derived_model and cfg.flash_attention:
+    if cfg.is_llama_derived_model and cfg.flash_attention:
         if cfg.device not in ["mps", "cpu"] and inference is False:
             from axolotl.flash_attn import replace_llama_attn_with_flash_attn
 
             logging.info("patching with flash attention")
             replace_llama_attn_with_flash_attn()
-    elif is_llama_derived_model and cfg.xformers_attention:
+    elif cfg.is_llama_derived_model and cfg.xformers_attention:
         from axolotl.monkeypatch.llama_attn_hijack_xformers import (
             hijack_llama_attention,
         )
 
         logging.info("patching with xformers attention")
         hijack_llama_attention()
-    elif is_llama_derived_model and cfg.sdp_attention:
+    elif cfg.is_llama_derived_model and cfg.sdp_attention:
         from axolotl.monkeypatch.llama_attn_hijack_xformers import (
             hijack_llama_sdp_attention,
         )
 
         logging.info("patching with sdp attention")
         hijack_llama_sdp_attention()
+    elif cfg.is_llama_derived_model and cfg.landmark_attention:
+        from axolotl.monkeypatch.llama_landmark_attn import (  # pylint: disable=redefined-outer-name # noqa: F811
+            MEM_TOKEN,
+            LlamaForCausalLM,
```

Comment:
reassigning this here inside the function causes this error:
```
ERROR:root:local variable 'LlamaForCausalLM' referenced before assignment
Traceback (most recent call last):
  File "/workspace/axolotl/src/axolotl/utils/models.py", line 214, in load_model
    model = LlamaForCausalLM.from_pretrained(
UnboundLocalError: local variable 'LlamaForCausalLM' referenced before assignment
^CTraceback (most recent call last):
  File "/workspace/axolotl/src/axolotl/utils/models.py", line 214, in load_model
    model = LlamaForCausalLM.from_pretrained(
UnboundLocalError: local variable 'LlamaForCausalLM' referenced before assignment
```

Reply:
Hm, weirdly I didn't get this error.

Second, the new PR should fix this import issue. But, it hasn't been tested.

Reply:
I need to go afk for now, so I'll test the PR when I get back since it seem like a big bug.